### PR TITLE
closes #58

### LINKED ISSUE #58: Allow nullable foreign keys in csvw
We want to allow foreign keys in tables to be optional, marked by a NULL value of the referencing column(s). The current behaviour can still be had by making the column(s) `required`.

### PR DESCRIPTION
**Pull request by xrotwang**
closes #58

### CODE REVIEW

File: clldutils/csvw/metadata.py [hunk 1]
```diff
@@ -620,7 +620,7 @@ def check_referential_integrity(self, data=None, log=None):
                     else:
                         colrefs = [colref]
                     for colref in colrefs:
-                        if colref not in keys:
+                        if not (all(c is None for c in colref) or colref in keys):
```
Comment (xflr6) [inline clldutils/csvw/metadata.py]: nit: how about ``if any(c is not None for c in colref) and colref not in keys``

File: clldutils/tests/test_csvw_metadata.py [hunk 2]
```diff
@@ -440,7 +440,7 @@ def test_foreignkeys(self):
 AF,33.9,67.7,Afghanistan""",
             "country_slice.csv": """\
 countryRef,year,population
-AF,1960,9616353
```
Comment (xflr6) [inline clldutils/tests/test_csvw_metadata.py]: just to make sure this is intended

File: clldutils/csvw/metadata.py (hunk 1 above)
Comment (xrotwang) [reply clldutils/csvw/metadata.py]: ok

File: clldutils/tests/test_csvw_metadata.py (hunk 2 above)
Comment (xrotwang) [reply clldutils/tests/test_csvw_metadata.py]: yes

File: clldutils/csvw/metadata.py (hunk 1 above)
Comment (xrotwang) [reply clldutils/csvw/metadata.py]: your version is 3 characters longer, though.

**Pull request merged**